### PR TITLE
Add hires TGI driver for the VIC.

### PR DESCRIPTION
**Pull request by SvenMichaelKlose**

### CODE REVIEW

File: libsrc/vic20/tgi_stat_stddrv.s [hunk 1]
```diff
@@ -0,0 +1,14 @@
+;
+; Address of the static standard tgi driver
+;
+; Oliver Schmidt, 2012-11-01
```
Comment (oliverschmidt) [inline libsrc/vic20/tgi_stat_stddrv.s]: Please don't C&P my name.

File: libsrc/vic20/tgi_stddrv.s [hunk 2]
```diff
@@ -0,0 +1,13 @@
+;
+; Name of the standard tgi driver
+;
+; Oliver Schmidt, 2011-05-02
```
Comment (oliverschmidt) [inline libsrc/vic20/tgi_stddrv.s]: Please don't C&P my name.

**Comment by oliverschmidt** [comment]
You need something like https://github.com/cc65/cc65/blob/master/include/c64.h#L83-L100

You need something like https://github.com/cc65/cc65/blob/master/include/c64.h#L159

Stupid question: How did you test your code without the stuff above in https://github.com/cc65/cc65/blob/master/include/vic20.h ?

File: include/vic20.h [hunk 3]
```diff
@@ -108,5 +130,7 @@ extern void vic20_stdjoy_joy[];         /* Referred to by joy_static_stddrv[] */
 extern void vic20_rama_emd[];
 extern void vic20_georam_emd[];
 
+extern void c64_hi_tgi[];
```
Comment (polluks) [inline include/vic20.h]: not literally ;-)

File: libsrc/vic20/tgi/vic20-hi.s [hunk 4]
```diff
@@ -0,0 +1,1054 @@
...
+        .include        "tgi-error.inc"
+
+        .macpack        generic
+        .macpack        module
+
+
+; ------------------------------------------------------------------------
+; Constants
+
+COLS := 20
```
Comment (oliverschmidt) [inline libsrc/vic20/tgi/vic20-hi.s]: Numeric constants are supposed to be set via `=`. See https://cc65.github.io/doc/ca65.html#ss6.1

File: libsrc/vic20/tgi/vic20-hi.s [hunk 5]
```diff
@@ -0,0 +1,1054 @@
...
+        .byte   $74, $67, $69           ; "tgi"
+        .byte   TGI_API_VERSION         ; TGI API version number
+        .addr   $0000                   ; Library reference
+        .word   XRES                    ; X resolution
+        .word   YRES                    ; Y resolution
+        .byte   2                       ; Number of drawing colors
+        .byte   1                       ; Number of screens available
+        .byte   8                       ; System font X size
+        .byte   8                       ; System font Y size
+        .word   $00C5                   ; Aspect ratio 2.5:3
```
Comment (oliverschmidt) [inline libsrc/vic20/tgi/vic20-hi.s]: Can you please elaborate how you get to this number? I'd rather see something like `$0198`.

**Comment by oliverschmidt** [comment]
I'm unfortunately pretty lost in understanding which part of the discussion in https://github.com/cc65/cc65/pull/611 applies here too...

**Comment by SvenMichaelKlose** [comment]
Job done by Greg King already. https://github.com/cc65/cc65/pull/611

**Pull request closed without merging**